API/socket.js

```diff
@@ -123,9 +123,48 @@ io.on('connection', (socket) => {
 	});
 });
 
-const PORT = process.env.APISOCKETPORT;
-const HOST = '0.0.0.0';
-
-logs.info('server listening on host ', HOST, 'port', PORT);
+const APISOCKETPORT = process.env.APISOCKETPORT;
+const APIHOST = process.env.APIHOST || '0.0.0.0';
+/**
+ * start the socket server and start listening
+ *
+ * @return     {Promise}  { resolves with { port , host} when listening }
+ */
+function listen() {
+	return new Promise((resolve, reject) => {
+		if (!APISOCKETPORT || !APIHOST) {
+			return reject('no APISOCKETPORT defined in environment');
```

Comment:
I don't think we need to return the reject here

Reply:
Should we also log this?

Reply:
done